### PR TITLE
Use jboss tattletale plugin

### PR DESCRIPTION
**Pull request by jdillon**
Use official jboss tattletale maven plugin.

Includes a bunch more reports too.  Configuration has the same excludes (which we really should fix).  Changed where the profile for this exists, its now a profile in nexus/pom.xml to enable the module, vs a profile in the module itself.

### CODE REVIEW

File: nexus/pom.xml [hunk 1]
```diff
@@ -1788,13 +1802,21 @@
...
     <module>nexus-stories</module>
     <module>nexus-test</module>
     <module>nexus-test-harness</module>
   </modules>
 
   <profiles>
+    <profile>
+      <id>tattletale</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
```
Comment (jdillon) [inline nexus/pom.xml]: May actually want to make this only run on grid things and for releases, not always run.

Comment (peterlynch) [inline nexus/pom.xml]: would like to avoid 'activeByDefault' for any profile.

Instead propose we add this to 'it' profile

Comment (jdillon) [inline nexus/pom.xml]: agree

**Comment by peterlynch** [comment]
+1

**Comment by nabcos** [comment]
+1

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
+1

**Comment by peterlynch** [comment]
added note to not use the other one

https://github.com/sonatype/tattletale-maven-plugin/blob/master/README.md

**Pull request merged**